packages/@glimmer/integration-tests/test/strict-mode-test.ts

```diff
@@ -904,31 +978,15 @@ class DynamicStrictModeTest extends RenderTest {
     const foo = defineComponent({}, 'Hello, world!');
     const Bar = defineComponent(
       {},
-      '{{#if (foo)}}{{/if}}',
-      class extends GlimmerishComponent {
-        foo = foo;
-      }
-    );
-
-    this.assert.throws(() => {
-      this.renderComponent(Bar);
-    }, /Attempted to resolve a helper in a strict mode template, but that value was not in scope: foo/);
-  }
-
-  @test
-  'Throws an error if a non-modifier is used as a modifier'() {
-    const foo = defineSimpleHelper(() => 'Hello, world!');
-    const Bar = defineComponent(
-      {},
-      '<div {{foo}}></div>',
+      '{{#if (this.foo)}}{{/if}}',
       class extends GlimmerishComponent {
         foo = foo;
       }
     );
 
     this.assert.throws(() => {
       this.renderComponent(Bar);
-    }, /Error: Attempted to resolve a modifier in a strict mode template, but it was not in scope: foo/);
+    }, /Expected a dynamic helper definition, but received an object or function that did not have a helper manager associated with it. The dynamic invocation was `{{this.foo}}` or `\(this.foo\)`, and the incorrect definition is the value at the path `this.foo`, which was:/);
```

Comment:
Realized this test was incorrect and fixed it tangentially, not directly related to this PR.